File: nodegs_cs/node.gs/Service/UserSecurityService.cs
using System;
using System.Collections.Generic;
using System.Text;
using node.gs;

namespace OwLibSV
{
    /// <summary>
    /// 自选股类别
    /// </summary>
    public class UserSecurityCategory
    {
        #region Lord 2016/12/22
        /// <summary>
        /// 类别ID
        /// </summary>
        public String m_categoryID = "";

        /// <summary>
        /// 股票代码
        /// </summary>
        public String m_codes = "";

        /// <summary>
        /// 类别名称
        /// </summary>
        public String m_name = "";

        /// <summary>
        /// 列表顺序
        /// </summary>
        public int m_orderNum;

        /// <summary>
        /// 类别
        /// </summary>
        public int m_type;

        /// <summary>
        /// 用户ID
        /// </summary>
        public int m_userID;
        #endregion
    }

    /// <summary>
    /// 自选股服务
    /// </summary>
    public class UserSecurityService : HttpHardService
    {
        #region Lord 2016/12/19
        /// <summary>
        /// 创建自选股服务
        /// </summary>
        public UserSecurityService()
        {
            ServiceID = SERVICEID_USERSECURITY;
        }

        /// <summary>
        /// 自选股服务ID
        /// </summary>
        public const int SERVICEID_USERSECURITY = 1;

        /// <summary>
        /// 添加类别方法ID
        /// </summary>
        public const int FUNCTIONID_USERSECURITY_ADDCATEGORIES = 0;

        /// <summary>
        /// 删除类别方法ID
        /// </summary>
        public const int FUNCTIONID_USERSECURITY_DELETECATEGORIES = 1;

        /// <summary>
        /// 获取所有类别信息方法ID
        /// </summary>
        public const int FUNCTIONID_USERSECURITY_GETCATEGORIES = 2;

        /// <summary>
        /// 更新类别信息方法ID
        /// </summary>
        public const int FUNCTIONID_USERSECURITY_UPDATECATEGORIES = 3;

        /// <summary>
        /// 添加证券方法ID
        /// </summary>
        public const int FUNCTIONID_USERSECURITY_ADDSECURITIES = 4;

        /// <summary>
        /// 删除证券方法ID
        /// </summary>
        public const int FUNCTIONID_USERSECURITY_DELETESECURITIES = 5;


        private int m_socketID = 0;

        /// <summary>
        /// 获取或设置套接字ID
        /// </summary>
        public int SocketID
        {
            get { return m_socketID; }
            set { m_socketID = value; }
        }

        /// <summary>
        /// 添加自选股类别
        /// </summary>
        /// <param name="requestID">请求ID</param>
        /// <param name="category">类别</param>
        /// <returns>状态</returns>
        public int AddCategory(int requestID, UserSecurityCategory category)
        {
            List<UserSecurityCategory> categories = new List<UserSecurityCategory>();
            categories.Add(category);
            int ret = Send(FUNCTIONID_USERSECURITY_ADDCATEGORIES, requestID, categories);
            categories.Clear();
            return ret > 0 ? 1 : 0;
        }

        /// <summary>
        /// 添加证券
        /// </summary>
        /// <param name="requestID">请求ID</param>
        /// <param name="category">类别</param>
        /// <returns>状态</returns>
        public int AddSecurities(int requestID, UserSecurityCategory category)
        {
            List<UserSecurityCategory> categories = new List<UserSecurityCategory>();
            categories.Add(category);
            int ret = Send(FUNCTIONID_USERSECURITY_ADDSECURITIES, requestID, categories);
            categories.Clear();
            return ret > 0 ? 1 : 0;
        }

        /// <summary>
        /// 删除类别
        /// </summary>
        /// <param name="requestID">请求ID</param>
        /// <param name="category">类别</param>
        /// <returns>状态</returns>
        public int DeleteCategory(int requestID, UserSecurityCategory category)
        {
            List<UserSecurityCategory> categories = new List<UserSecurityCategory>();
            categories.Add(category);
            int ret = Send(FUNCTIONID_USERSECURITY_DELETECATEGORIES, requestID, categories);
            categories.Clear();
            return ret > 0 ? 1 : 0;
        }

        /// <summary>
        /// 删除证券
        /// </summary>
        /// <param name="requestID">请求ID</param>
        /// <param name="category">类别</param>
        /// <returns>状态</returns>
        public int DeleteSecurities(int requestID, UserSecurityCategory category)
        {
            List<UserSecurityCategory> categories = new List<UserSecurityCategory>();
            categories.Add(category);
            int ret = Send(FUNCTIONID_USERSECURITY_DELETESECURITIES, requestID, categories);
            categories.Clear();
            return ret > 0 ? 1 : 0;
        }

        /// <summary>
        /// 获取类别信息
        /// </summary>
        /// <param name="requestID">请求ID</param>
        /// <param name="category">类别</param>
        /// <returns>状态</returns>
        public int GetCategories(int requestID, UserSecurityCategory category)
        {
            List<UserSecurityCategory> categories = new List<UserSecurityCategory>();
            categories.Add(category);
            int ret = Send(FUNCTIONID_USERSECURITY_GETCATEGORIES, requestID, categories);
            categories.Clear();
            return ret > 0 ? 1 : 0;
        }

        /// <summary>
        /// 通过流获取类别信息列表
        /// </summary>
        /// <param name="categories">类别信息</param>
        /// <param name="body">包体</param>
        /// <param name="bodyLength">包体长度</param>
        /// <returns>状态</returns>
        public static int GetCategories(List<UserSecurityCategory> categories, byte[] body, int bodyLength)
        {
            Binary br = new Binary();
            br.Write(body, bodyLength);
            int size = br.ReadInt();
            if (size > 0)
            {
                for (int i = 0; i < size; i++)
                {
                    UserSecurityCategory category = new UserSecurityCategory();
                    category.m_categoryID = br.ReadString();
                    category.m_userID = br.ReadInt();
                    category.m_name = br.ReadString();
                    category.m_codes = br.ReadString();
                    category.m_type = br.ReadInt();
                    category.m_orderNum = br.ReadInt();
                    categories.Add(category);
                }
            }
            br.Close();
            return 1;
        }

        /// <summary>
        /// 接收数据
        /// </summary>
        /// <param name="message">消息</param>
        public override void OnReceive(CMessage message)
        {
            base.OnReceive(message);
            SendToListener(message);
        }

        /// <summary>
        /// 发送消息
        /// </summary>
        /// <param name="functionID">方法ID</param>
        /// <param name="requestID">请求ID</param>
        /// <param name="categories">类别集合</param>
        /// <returns>状态</returns>
        public int Send(int functionID, int requestID, List<UserSecurityCategory> categories)
        {
            Binary bw = new Binary();
            int categoriesSize = categories.Count;
            bw.WriteInt(categoriesSize);
            for (int i = 0; i < categoriesSize; i++)
            {
                UserSecurityCategory category = categories[i];
                bw.WriteString(category.m_categoryID);
                bw.WriteInt(category.m_userID);
                bw.WriteString(category.m_name);
                bw.WriteString(category.m_codes);
                bw.WriteInt(category.m_type);
                bw.WriteInt(category.m_orderNum);
            }
            byte[] bytes = bw.GetBytes();
            int ret = Send(new CMessage(GroupID, ServiceID, functionID, SessionID, requestID, m_socketID, 0, CompressType, bytes.Length, bytes));
            bw.Close();
            return ret;
        }

        /// <summary>
        /// 更新类别信息
        /// </summary>
        /// <param name="requestID">请求ID</param>
        /// <param name="category">类别</param>
        /// <returns>状态</returns>
        public int UpdateCategory(int requestID, UserSecurityCategory category)
        {
            List<UserSecurityCategory> categories = new List<UserSecurityCategory>();
            categories.Add(category);
            int ret = Send(FUNCTIONID_USERSECURITY_UPDATECATEGORIES, requestID, categories);
            categories.Clear();
            return ret > 0 ? 1 : 0;
        }

        ///// <summary>
        ///// 更新指标的排序
        ///// </summary>
        ///// <param name="requestID">请求ID</param>
        ///// <param name="userID">用户ID</param>
        ///// <param name="ids">ID</param>
        ///// <returns>状态</returns>
        //public int UpdateOrderNum(int requestID, int userID, List<String> ids)
        //{
        //    String str = "";
        //    int idsSize = ids.Count;
        //    for (int i = 0; i < idsSize; i++)
        //    {
        //        str += ids[i];
        //        if (i != idsSize - 1)
        //        {
        //            str += ",";
        //        }
        //    }
        //    UserSecurityCategory category = new UserSecurityCategory();
        //    category.m_categoryID = str;
        //    category.m_userID = userID;
        //    List<UserSecurityCategory> categories = new List<UserSecurityCategory>();
        //    categories.Add(category);
        //    int ret = Send(FUNCTIONID_USERSECURITY_UPDTAEORDERNUM, requestID, categories);
        //    categories.Clear();
        //    return ret > 0 ? 1 : 0;
        //}
        #endregion
    }
}

File: nodegs_java/node.gs/src/nodegs/service/UserSecurityService.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package nodegs.service;

import com.mongodb.*;
import gaia.owchart.owchart_android.Base.CStr;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;
import java.util.UUID;
import nodegs.base.Binary;
import nodegs.base.CMessage;
import nodegs.base.HttpHardService;

/**
 *
 * @author GAIA
 */
public class UserSecurityService extends HttpHardService
{   
    /*
        创建服务
    */
    public UserSecurityService()
    {
        SetServiceID(SERVICEID_USERSECURITY);
    }
        
    /**
     * 自选股服务ID
     */
    public final int SERVICEID_USERSECURITY = 1;

    /**
     * 添加类别方法ID
     */
    public final int FUNCTIONID_USERSECURITY_ADDCATEGORIES = 0;

    /**
     * 删除类别方法ID
     */
    public final int FUNCTIONID_USERSECURITY_DELETECATEGORIES = 1;

    /**
     * 获取所有类别信息方法ID
     */
    public final int FUNCTIONID_USERSECURITY_GETCATEGORIES = 2;

    /**
     * 更新类别信息方法ID
     */
    public final int FUNCTIONID_USERSECURITY_UPDATECATEGORIES = 3;

    /**
     * 添加证券方法ID
     */
    public final int FUNCTIONID_USERSECURITY_ADDSECURITIES = 4;

    /**
     * 删除证券方法ID
     */
    public final int FUNCTIONID_USERSECURITY_DELETESECURITIES = 5;

    /**
     * 更新类别排序方法ID
     */
    public final int FUNCTIONID_USERSECURITY_UPDTAEORDERNUM = 6;

    public final String MONGODB_USER_CATAGORY = "UserCatagory";

    private MongoService m_mongoService;
        
    public DBCollection GetCollection()
    {
        DBCollection collection = m_mongoService.GetCollectionByName(MONGODB_USER_CATAGORY);
        return collection;
    }
        
    /**
     * 添加自选股类别
     * @param message 消息
     * @return  状态
     */
    public int AddCategories(CMessage message)
    {
        ArrayList<UserSecurityCategory> categories = new ArrayList<UserSecurityCategory>();
        GetCategories(categories, message.m_body, message.m_bodyLength);
        int categoriesSize = categories.size();
        ArrayList<UserSecurityCategory> addCategories = new ArrayList<UserSecurityCategory>();
        for (int i = 0; i < categoriesSize; i++)
        {
            UserSecurityCategory receive = categories.get(i);
            receive.m_categoryID = UUID.randomUUID().toString();
            
            GetCollection().insert(UserSecurityCategory2DBObject(receive));
            
            addCategories.add(receive);
        }
        int ret = Send(message, addCategories);
        categories.clear();
        addCategories.clear();
        return ret;
    }
        
    /**
     * 添加证券
     * @param message 消息
     * @return 状态
     */
    public int AddSecurities(CMessage message)
    {
        UserSecurityCategory receive = new UserSecurityCategory();
        ArrayList<UserSecurityCategory> categories = new ArrayList<UserSecurityCategory>();
        GetCategories(categories, message.m_body, message.m_bodyLength);
        receive = categories.get(0);
        categories.clear();            
        GetCategories(categories, receive.m_userID, receive.m_categoryID);
        int categoriesSize = categories.size();
        if (categoriesSize > 0)
        {
            ArrayList<String> addedCodes = new ArrayList<String>();
            UserSecurityCategory category = categories.get(0);
            String[] newCodes = receive.m_codes.split(",");
            int newSize = newCodes.length;
            if (category.m_codes != null && category.m_codes.length() > 0)
            {
                String[] oldCodes = category.m_codes.split(",");
                HashMap<String, String> oldCodesMap = new HashMap<String, String>();
                int oldSize = oldCodes.length;
                for (int i = 0; i < oldSize; i++)
                {
                    oldCodesMap.put(oldCodes[i], "");
                }
                for (int i = 0; i < newSize; i++)
                {
                    if (!oldCodesMap.containsKey(newCodes[i]))
                    {
                        addedCodes.add(newCodes[i]);
                        category.m_codes += "," + newCodes[i];
                    }
                }
            }
            else
            {
                category.m_codes = receive.m_codes;
                for (int i = 0; i < newSize; i++)
                {
                    addedCodes.add(newCodes[i]);
                }
            }
            UpdateCategory(category);
            String sendCodes = "";
            int addedCodesSize = addedCodes.size();
            for (int i = 0; i < addedCodesSize; i++)
            {
                sendCodes += addedCodes.get(i);
                if (i != addedCodesSize - 1)
                {
                    sendCodes += ",";
                }
            }
            category.m_codes = sendCodes;
            int ret = Send(message, categories);
            categories.clear();
            return ret;
        }
        return -1;
    }
        
    public static UserSecurityCategory DBObject2UserSecurityCategory(DBObject DBObject)
    {
        @SuppressWarnings("unchecked")
        Map<String, Object> userMap = (Map<String, Object>) DBObject.toMap();

        UserSecurityCategory userSecurityCategory = new UserSecurityCategory();  
        userSecurityCategory.m_categoryID = String.valueOf(userMap.get("m_categoryID"));
        userSecurityCategory.m_codes = String.valueOf(userMap.get("m_codes"));
        userSecurityCategory.m_name = String.valueOf(userMap.get("m_name"));
        userSecurityCategory.m_orderNum = Integer.valueOf(String.valueOf(userMap.get("m_orderNum")));
        userSecurityCategory.m_type = Integer.valueOf(String.valueOf(userMap.get("m_type")));
        userSecurityCategory.m_userID = Integer.valueOf(String.valueOf(userMap.get("m_userID")));
        return userSecurityCategory;
    }

    /**
     * 删除类别
     * @param message 消息
     * @return 状态
     */
    public int DeleteCategories(CMessage message)
    {
        ArrayList<UserSecurityCategory> categories = new ArrayList<UserSecurityCategory>();
        GetCategories(categories, message.m_body, message.m_bodyLength);
        int categoriesSize = categories.size();
        ArrayList<UserSecurityCategory> deleteCategories = new ArrayList<UserSecurityCategory>();
        for (int i = 0; i < categoriesSize; i++)
        {
            UserSecurityCategory receive = categories.get(i);    

            GetCollection().remove(UserSecurityCategory2DBObject(receive));
            
            deleteCategories.add(receive);
        }
        int ret = Send(message, deleteCategories);
        categories.clear();
        deleteCategories.clear();
        return ret;
    }

    /**
     * 删除证券
     * @param message
     * @return 
     */
    public int DeleteSecurities(CMessage message)
    {         
        UserSecurityCategory receive = new UserSecurityCategory(); 
        ArrayList<UserSecurityCategory> categories = new ArrayList<UserSecurityCategory>();
        GetCategories(categories, message.m_body, message.m_bodyLength);
        int categoriesSize = categories.size();
        ArrayList<UserSecurityCategory> deleteCategories = new ArrayList<UserSecurityCategory>();
         if (categoriesSize > 0)
        {
            UserSecurityCategory category = categories.get(0);
            ArrayList<String> deletedCodes = new ArrayList<String>();
            if (category.m_codes != null && category.m_codes.length() > 0)
            {
                String[] newCodes = receive.m_codes.split(",");
                String[] oldCodes = category.m_codes.split(",");
                HashMap<String, String> oldCodesMap = new HashMap<String, String>();
                HashMap<String, String> newCodesMap = new HashMap<String, String>();
                int oldSize = oldCodes.length;
                for (int i = 0; i < oldSize; i++)
                {
                    oldCodesMap.put(oldCodes[i], "");
                }
                int newSize = newCodes.length;
                for (int i = 0; i < newSize; i++)
                {
                    newCodesMap.put(newCodes[i], "");
                }
                for (String code : newCodesMap.keySet())
                {
                    if (oldCodesMap.containsKey(code))
                    {
                        deletedCodes.add(code);
                        oldCodesMap.remove(code);
                    }
                }
                int pos = 0;
                oldSize = oldCodesMap.size();
                String newSaveCodes = "";
                for (String code : oldCodesMap.keySet())
                {
                    newSaveCodes += code;
                    if (pos != oldSize - 1)
                    {
                        newSaveCodes += ",";
                    }
                    pos++;
                }
                category.m_codes = newSaveCodes;
            }
            UpdateCategory(category);
            String sendCodes = "";
            int deletedCodesSize = deletedCodes.size();
            for (int i = 0; i < deletedCodesSize; i++)
            {
                sendCodes += deletedCodes.get(i);
                if (i != deletedCodesSize - 1)
                {
                    sendCodes += ",";
                }
            }
            category.m_codes = sendCodes;
            int ret = Send(message, categories);
            categories.clear();
            return ret;
        }
        return -1;
    }
        
    /**
     *  获取类别信息
     * @param categories 类别
     * @param body 包体
     * @param bodyLength  包体长度
     */
    private void GetCategories(ArrayList<UserSecurityCategory> categories, byte[] body, int bodyLength)
    {
        try
        {
            Binary br = new Binary();
            br.Write(body, bodyLength);
            int categoriesSize = br.ReadInt();
            for (int i = 0; i < categoriesSize; i++)
            {
                UserSecurityCategory category = new UserSecurityCategory();
                category.m_categoryID = br.ReadString();
                category.m_userID = br.ReadInt();
                category.m_name = br.ReadString();
                category.m_codes = br.ReadString();
                category.m_type = br.ReadInt();
                category.m_orderNum = br.ReadInt();
                if (category.m_categoryID == "primary")
                {
                    category.m_categoryID = GetPrimaryCategoryID(category.m_userID);
                    if (category.m_categoryID.length() > 0)
                    {
                        categories.add(category);
                    }
                }
                else
                {
                    categories.add(category);
                }
            }
            br.Close();                
        }
        catch(Exception ex)
        {
        }            
    }
        
    /**
     *  获取类别信息
     * @param categories 类别信息
     * @param userID 用户ID
     * @param categoryID 类别ID
     * @return  状态
     */
    public int GetCategories(ArrayList<UserSecurityCategory> categories, int userID, String categoryID)
    {
        DBCollection userCollection = GetCollection();
        BasicDBObject query =new BasicDBObject();
        if(categoryID.length() != 0)
        {
            query.put("m_categoryID", categoryID);
        }
        UserSecurityCategory category = null;

        ArrayList<UserSecurityCategory> allCategory = new ArrayList<UserSecurityCategory>();
        DBCursor cursor = userCollection.find(query);
        while(cursor.hasNext())
        {
            DBObject dbObject = cursor.next();            
            category = DBObject2UserSecurityCategory(dbObject);
            allCategory.add(category);              
        }
        
        int size = allCategory.size();
        if (categoryID != null && categoryID.length() > 0)
        {
            for (int i = 0; i < size; i++)
            {
                if (allCategory.get(i).m_categoryID.equals(categoryID))
                {
                    categories.add(allCategory.get(i));
                }
            }           
        }
        else
        {
            if (userID > 0)
            {
                for (int i = 0; i < size; i++)
                {
                    if (allCategory.get(i).m_userID == userID)
                    {
                        categories.add(allCategory.get(i));
                    } 
                }
            }
            else
            {
                categories = allCategory;
            }            
        }
        return 1;
    }
        
    /**
     * 获取所有类别信息
     * @param message 消息
     * @return 状态
     */
    public int GetCategories(CMessage message)
    {
        UserSecurityCategory receive = new UserSecurityCategory();
        ArrayList<UserSecurityCategory> categories = new ArrayList<UserSecurityCategory>();
        GetCategories(categories, message.m_body, message.m_bodyLength);
        receive = categories.get(0);
        categories.clear();
        GetCategories(categories, receive.m_userID, receive.m_categoryID);
        int ret = Send(message, categories);
        return ret;
    }
        
    /**
     * 获取主要的类别ID
     * @param userID 用户ID
     * @return 类别ID
     */
    public String GetPrimaryCategoryID(int userID)
    {
        String categoryID = "111111";
        return categoryID;
    }

    @Override
    public void OnReceive(CMessage message)
    {
        super.OnReceive(message);
        switch (message.m_functionID)
        {
            case FUNCTIONID_USERSECURITY_ADDCATEGORIES:
                AddCategories(message);
                break;
            case FUNCTIONID_USERSECURITY_DELETECATEGORIES:
                DeleteCategories(message);
                break;
            case FUNCTIONID_USERSECURITY_GETCATEGORIES:
                GetCategories(message);
                break;
            case FUNCTIONID_USERSECURITY_UPDATECATEGORIES:
                UpdateCategories(message);
                break;
            case FUNCTIONID_USERSECURITY_ADDSECURITIES:
                AddSecurities(message);
                break;
            case FUNCTIONID_USERSECURITY_DELETESECURITIES:
                DeleteSecurities(message);
                break;
            case FUNCTIONID_USERSECURITY_UPDTAEORDERNUM:
                UpdateOrderNum(message);
                break;
        }
    }

    /// <summary>
    /// 发送类别
    /// </summary>
    /// <param name="message">消息</param>
    /// <param name="categories">类别列表</param>
    public int Send(CMessage message, ArrayList<UserSecurityCategory> categories)
    {
        try
        {
            Binary bw = new Binary();
            int size = categories.size();
            bw.WriteInt(size);
            for (int i = 0; i < size; i++)
            {
                UserSecurityCategory category = categories.get(i);
                bw.WriteString(category.m_categoryID);
                bw.WriteInt(category.m_userID);
                bw.WriteString(category.m_name);
                bw.WriteString(category.m_codes);
                bw.WriteInt(category.m_type);
                bw.WriteInt(category.m_orderNum);
            }
            byte[] bytes = bw.GetBytes();
            message.m_bodyLength = bytes.length;
            message.m_body = bytes;
            int ret = Send(message);
            bw.Close();
            return ret;
        }
        catch(Exception ex)
        {
        }
        return 1;
    }

    /**
     *  更新类别
     * @param message 消息
     * @return 状态
     */
    public int UpdateCategories(CMessage message)
    {
        ArrayList<UserSecurityCategory> categories = new ArrayList<UserSecurityCategory>();
        GetCategories(categories, message.m_body, message.m_bodyLength);
        int categoriesSize = categories.size();
        ArrayList<UserSecurityCategory> updateCategories = new ArrayList<UserSecurityCategory>();
        for (int i = 0; i < categoriesSize; i++)
        {
            UserSecurityCategory receive = categories.get(i);
            UpdateCategory(receive);
            updateCategories.add(receive);
        }
        int ret = Send(message, updateCategories);
        categories.clear();
        updateCategories.clear();
        return ret;
    }

    /**
     * 更新类别信息
     * @param category 类别信息
     */
    public void UpdateCategory(UserSecurityCategory category)
    {
        GetCollection().update(new BasicDBObject().append("m_categoryID", category.m_categoryID), UserSecurityCategory2DBObject(category));
    }

    /**
     * 更新排序字段
     * @param message 消息
     * @return 状态
     */
    public int UpdateOrderNum(CMessage message)
    {
        ArrayList<UserSecurityCategory> categories = new ArrayList<UserSecurityCategory>();
        GetCategories(categories, message.m_body, message.m_bodyLength);
        int categoriesSize = categories.size();
        ArrayList<UserSecurityCategory> updateCategories = new ArrayList<UserSecurityCategory>();
        for (int i = 0; i < categoriesSize; i++)
        {
            UserSecurityCategory receive = categories.get(i);
            String[] ids = receive.m_categoryID.split(",");
            int idsSize = ids.length;
            for (int j = 0; j < idsSize; j++)
            {
                receive.m_orderNum = j;

                GetCollection().insert(UserSecurityCategory2DBObject(receive));
            }
            updateCategories.add(receive);
        }
        int ret = Send(message, updateCategories);
        categories.clear();
        updateCategories.clear();
        return ret;
    }    
        
    public static BasicDBObject UserSecurityCategory2DBObject(UserSecurityCategory category)
    {
        BasicDBObject object = new BasicDBObject();
        object.put("m_categoryID", category.m_categoryID);
        object.put("m_codes", category.m_codes);
        object.put("m_name", category.m_name);
        object.put("m_orderNum", category.m_orderNum);
        object.put("m_type", category.m_type);
        object.put("m_userID", category.m_userID);
        return object;
    }
}
